src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -402,6 +424,7 @@ void PDPServer::assignRemoteEndpoints(
 {
     EPROSIMA_LOG_INFO(RTPS_PDP_SERVER, "Assigning remote endpoint for RTPSParticipant: " << pdata->m_guid.guidPrefix);
 
+    auto endpoints = static_cast<fastdds::rtps::DiscoveryServerPDPEndpoints*>(builtin_endpoints_.get());
```

Comment:
Except for a couple of `dynamic_cast<>` operations which are needed upon stateful writer/readers creation (a `RTPSReader/Writer*` is returned), perhaps some of the `static_cast<DiscoveryServerPDPEndpoints> `calls could be saved up in this file, since it is costy and I find it redundant (I dont see now any external operation that can potentially change the type except for those above) neither the `builtin_endpoints_` member is exposed to be externally modified.

Reply:
> It can be that .get() from unique_ptr returns the raw pointer to the base class, so the cast is always needed

It is exactly that